src/lavinmq/federation/upstream_store.cr

```diff
@@ -1,15 +1,19 @@
 require "./upstream"
+require "../logging"
 
 module LavinMQ
   module Federation
     class UpstreamStore
       include Enumerable(Upstream)
+
+      Log = Federation::Log.for "upstreamstore"
+
       @upstreams = Hash(String, Upstream).new
       @upstream_sets = Hash(String, Array(Upstream)).new
-      @log : Log
+      @log : Logging::EntityLog
 
       def initialize(@vhost : VHost)
-        @log = Log.for "UpstreamStore[vhost=#{@vhost}]"
+        @log = @vhost.log.extend(Log)
```

Comment:
Yes, absolutely.

IIRC the reason for setting `Log` and use that was that I assumed that `Log.for` would create a new `Log` instance for every call. Now that I've read the code and I see that it's not, this could be changed.